<div align="center">

**LAPORAN PRAKTIKUM**

**ALGORITMA DAN PEMROGRAMAN**

**“TYPE DATA DAN VARIABEL”**

**Disusun Oleh :**

**Dika Giowanda**

**2511533025**

**Dosen Pengampu:**

**Wahyudi.DR.. S.T.M.T**

**Asisten Praktikum:**

**Muhammad Galid Avero**

<br>

 <img src="images/Logo_unand.png" width="200">

<br>

**FAKULTAS TEKNOLOGI INFORMASI**

**DEPARTEMEN INFORMATIKA**

**UNIVERSITAS ANDALAS**

**2026**

</div>

## KATA PENGANTAR

Puji syukur penulis panjatkan atas kehadirat Tuhan Yang Maha Esa yang telah memberikan rahmat dan berkatnya, sehingga penulis dapat menyelesaikan laporan praktikum Algoritma Dan Pemrograman yang telah dilaksanakan sebagai pertemuan ke-1 untuk mata kuliah Praktikum Algoritma Dan Pemrograman. Praktikum ini berfokus pada pembahasan tentang tipe data dan variabel.

Melalui praktikum ini, hasil yang diharapkan adalah penulis dapat memahami dan mengimplementasikan secara langsung konsep tipe data dan variabel pada real project.

Penulis menyadari bahwa laporan praktikum ini masih memiliki banyak kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan demi perbaikan dan pengembangan di masa mendatang.

Padang, 2026

---
# BAB I
# PENDAHULUAN

## 1.1 Latar Belakang
Dalam dunia pemrograman dan pengembangan perangkat lunak, variabel dan tipe data merupakan konsep paling mendasar yang harus dipahami oleh setiap programmer. Variabel berfungsi sebagai wadah di dalam memori komputer untuk menyimpan data yang dapat berubah selama program berjalan, sedangkan tipe data menentukan jenis nilai yang dapat disimpan serta operasi yang dapat dilakukan terhadap data tersebut.

Pemilihan tipe data yang tepat sangat berpengaruh terhadap performa program dan alokasi memori komputer secara keseluruhan. Ketidaktepatan dalam memilih tipe data atau mengelola variabel dapat menyebabkan terjadinya kesalahan logika, kegagalan sistem (crash), hingga inefisiensi memori yang berdampak buruk pada performa aplikasi saat dijalankan. Selain itu, pemahaman mendalam tentang sifat alokasi data di memori menjadi dasar utama sebelum melangkah ke topik struktur data yang lebih kompleks.

Pemahaman yang kuat mengenai tipe data dasar (seperti integer, float, string, dan boolean) serta deklarasi variabel sangat penting dalam menjaga efisiensi penggunaan memori dan mencegah terjadinya kesalahan (error) komputasi atau konversi tipe data. Oleh karena itu, praktikum ini dilaksanakan untuk memberikan pemahaman secara mendalam serta pengalaman langsung dalam menerapkan konsep variabel dan tipe data dalam penulisan program.

## 1.2 Tujuan
1. Memahami konsep dasar variabel, deklarasi, dan inisialisasi variabel dalam bahasa pemrograman.
2. Mengenal dan memahami berbagai jenis tipe data dasar (seperti integer, float, string, dan boolean).
3. Mampu menentukan dan mengimplementasikan tipe data yang tepat sesuai dengan kebutuhan data dalam pembuatan program.
4. Memahami konsep konversi tipe data (type casting) dan manipulasi variabel dalam proses pemrosesan data.
5. Mampu menyelesaikan permasalahan pemrograman sederhana dengan memanfaatkan variabel dan tipe data secara efektif.

## 1.3 Manfaat
1. Manfaat Bagi Praktikan:
    
    a. Memperkuat pemahaman logika dasar pemrograman sebagai fondasi sebelum mempelajari materi yang lebih kompleks seperti struktur kontrol dan struktur data. 
    
    b. Meningkatkan keterampilan dalam mendesain variabel yang efisien dan meminimalkan kesalahan (bug) akibat ketidaksesuaian tipe data.

2. Manfaat Bagi Pengembangan Program/Aplikasi:
    
    a. Membantu penyusunan kode program yang lebih terstruktur, mudah dibaca, dan efisien dalam penggunaan memori komputer. 
    
    b. Menjamin keakuratan proses manipulasi dan pengolahan data dalam suatu perangkat lunak.

---
# BAB II
# PEMBAHASAN

## 2.1 Dasar Teori
Dalam bidang algoritma dan pemrograman, pemrosesan data oleh komputer membutuhkan penyimpanan sementara dan aturan mengenai jenis data yang diolah. Konsep mendasar yang mengatur hal tersebut adalah variabel dan tipe data.
1. Pengertian Variabel

    Variabel adalah sebuah nama atau simbol yang digunakan untuk mengidentifikasi suatu lokasi penampungan data di dalam memori komputer (RAM). Nilai yang tersimpan dalam variabel dapat diakses, diubah, atau dimanipulasi selama program berlangsung.

2. Aturan Penamaan Variabel

    Dalam menuliskan nama variabel, terdapat beberapa aturan baku yang harus dipatuhi agar tidak terjadi syntax error:

    a. Nama variabel bersifat case sensitive (membedakan huruf besar dan huruf kecil, contoh: nilai berbeda dengan Nilai).

    b. Harus diawali dengan huruf atau karakter underscore (_), tidak boleh diawali dengan angka.

    c. Tidak boleh menggunakan kata kunci khusus dalam bahasa pemrograman (seperti if, for, class, import).

    d. Tidak boleh memuat spasi atau karakter khusus/simbol seperti @, #, $, %.

3. Deklarasi dan Inisialisasi Variabel

    a. Deklarasi: Proses mendaftarkan nama variabel beserta tipe datanya ke sistem agar memori dialokasikan.

    b. Inisialisasi: Proses pemberian nilai awal (initial value) pada variabel yang telah dideklarasikan.

4. Pengertian Tipe Data

    Tipe Data adalah klasifikasi yang memberitahu sistem atau interpreter mengenai jenis nilai yang dikandung oleh suatu variabel, bagaimana data tersebut disimpan di memori, serta operasi aritmatika atau logika apa saja yang dapat dilakukan terhadap data tersebut.

5. Klasifikasi Tipe Data Dasar (Primitive Data Types)

    a. Integer (Bilangan Bulat): Tipe data yang digunakan untuk menyimpan angka bulat positif maupun negatif tanpa koma/desimal (contoh: -5, 0, 42).

    b. Float / Double (Bilangan Desimal): Tipe data yang menyimpan angka pecahan atau memiliki koma desimal (contoh: 3.14, -0.001, 2.5).

    c. String (Teks / Karakter Kombinasi): Tipe data yang menyimpan sekumpulan karakter berupa teks, angka, maupun simbol yang diapit oleh tanda petik (contoh: "Hello World", "A123").

    d. Boolean (Nilai Logika): Tipe data sederhana yang hanya memiliki dua kemungkinan nilai, yaitu True (Benar) atau False (Salah), yang umumnya digunakan dalam percabangan logika.

    e. Character (Char): Tipe data yang digunakan untuk menyimpan tunggal satu karakter (contoh: 'A', 'z', '8').

6. Konversi Tipe Data (Type Casting)
    Konversi tipe data adalah proses mengubah nilai dari satu tipe data ke tipe data lainnya. Terdiri dari dua jenis:

    a.Implicit Casting (Otomatis): Konversi yang dilakukan secara otomatis oleh bahasa pemrograman ketika mengubah dari tipe data berkapasitas kecil ke besar.

    b. Explicit Casting (Manual): Konversi yang dilakukan secara sengaja oleh programmer menggunakan fungsi khusus (contoh: mengubah String "123" menjadi Integer 123).

7. Konstanta
    Konstanta adalah jenis variabel khusus yang nilainya bersifat tetap dan tidak dapat diubah setelah didefinisikan selama program dijalankan (contoh: nilai Pi = 3.14159).

8. Cakupan Variabel (Variable Scope)
    
    a. Variabel Lokal: Variabel yang dideklarasikan di dalam suatu fungsi/blok tertentu dan hanya bisa diakses di dalam blok tersebut.

    b. Variabel Global: Variabel yang dideklarasikan di luar fungsi/blok sehingga dapat diakses oleh seluruh bagian program.

## 2.2 Kode Program

### 2.2.1 Program "Numerik_2511533025"


In [1]:
# Buat file dengan nama Numerik_NIM.py
# Create a variable with integer value.
a_3025=100
print("The type of variable having value", a_3025, " is ", type(a_3025))

# Create a variable with float value.
b_3025=10.2345
print("The type of variable having value", b_3025, " is ", type(b_3025))

# Create a variable with complex value.
c_3025=100+3j
print("The type of variable having value", c_3025, " is ", type(c_3025))

The type of variable having value 100  is  <class 'int'>
The type of variable having value 10.2345  is  <class 'float'>
The type of variable having value (100+3j)  is  <class 'complex'>


**Penjelasan :**
1. Bagian pertama kode (Penggunaan Tipe Data Integer) 

    Berfungsi untuk menginisialisasi variabel a dengan nilai bilangan bulat 100. Fungsi type(a) digunakan untuk mengecek kelas tipe data dari variabel tersebut. Hasil eksekusi menampilkan <class 'int'>, yang menunjukkan bahwa variabel a menyimpan data berjenis Integer.

2. Bagian kedua kode (Penggunaan Tipe Data Float) 

    Berfungsi untuk menginisialisasi variabel b dengan nilai desimal 10.2345. Ketika fungsi type(b) dipanggil, program mengenali nilai ini sebagai bilangan pecahan/riil. Hasil eksekusi menampilkan <class 'float'>, yang menandakan bahwa variabel b memiliki tipe data Float (bilangan berkoma).

3. Bagian ketiga kode (Penggunaan Tipe Data Complex) 

    Berfungsi untuk mendefinisikan variabel c dengan nilai bilangan kompleks 100+3j, di mana 100 adalah bagian riil dan 3j adalah bagian imajiner (dalam Python digunakan huruf j). Hasil pemeriksaan fungsi type(c) menghasilkan <class 'complex'>, yang mengonfirmasi bahwa variabel c merupakan tipe data Complex (bilangan kompleks).

4. Penggunaan Fungsi print() dan type() 

    Berfungsi untuk menampilkan output teks gabungan antara pesan deskriptif, nilai variabel yang tersimpan, serta tipe data resmi yang terdeteksi oleh interpreter Python secara dinamis (dynamic typing).

### 2.2.2 Program "Boolean_2511533025"

In [1]:
# Buat file dengan nama Boolean_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: nilai_1234
# Deklarasi variabel dengan tipe data Boolean
is_lulus_3025 = True
is_cumlaude_3025 = True

# Menggunakan Boolean
nilai_3025 = 85
batas_lulus_3025 = 75

# Menentukan nilai Boolean dari kondisi
status_kelulusan_3025 = nilai_3025 >= batas_lulus_3025 # Hasilnya akan true

print("=== Check Kelulusan ===")
print("Nilai:", nilai_3025)
print("Apakah Lulus?:", status_kelulusan_3025)
if is_lulus_3025 and is_cumlaude_3025:
    print("Selamat, Anda lulus dengan predikat Cum Laude!")

=== Check Kelulusan ===
Nilai: 85
Apakah Lulus?: True
Selamat, Anda lulus dengan predikat Cum Laude!


**Penjelasan :**
1. Deklarasi Variabel Boolean secara Langsung

    Bagian ini mendefinisikan variabel is_lulus dan is_cumlaude yang bernilai True. Tipe data Boolean ini digunakan untuk menyimpan status kebenaran logika (True atau False).

2. Inisialisasi Variabel Numerik

    Bagian ini mendeklarasikan variabel nilai dengan nilai 85 dan batas_lulus dengan nilai 75. Kedua variabel ini memiliki tipe data Integer yang nantinya digunakan sebagai acuan perbandingan logika.

3. Penentuan Nilai Boolean dari Evaluasi Kondisi

    Variabel status_kelulusan menyimpan hasil perbandingan ekspresi logika nilai >= batas_lulus. Karena nilai 85 lebih besar atau sama dengan 75, maka ini mengevaluasi ke nilai True, yang otomatis membuat status_kelulusan bertipe data Boolean dengan nilai True.

4. Menampilkan Informasi (Pencetakan Output)

    Fungsi print() digunakan untuk menampilkan teks, nilai dari variabel nilai (yaitu 85), serta hasil evaluasi logika pada status_kelulusan (yaitu True).

5. Pengujian Kondisi Logika Terstruktur

    Struktur percabangan if mengevaluasi gabungan dua variabel Boolean menggunakan operator logika and (is_lulus and is_cumlaude). Karena kedua variabel tersebut bernilai True, maka secara keseluruhan dan blok kode di dalamnya mengeksekusi perintah untuk mencetak pesan "Selamat, Anda lulus dengan predikat Cum Laude!".

### 2.2.3 Program "String_2511533025"

In [1]:
# Buat file dengan nama String_NIM.py
# Nama variable ditambah 4 digit nim terakhir contoh: nama_1234
# Ganti programnya dan data masing2
nama_3025 = "Dika Giowanda"
jenis_kelamin_3025 = 'L'
alamat_3025 = """
    Padang Lawas
    kec. Luhak Nan Duo
    Kab. Pasaman Barat
"""
agama_3025 = 'islam'
print("Nama saya adalah: ", nama_3025)
print("Jenis kelamin saya adalah: ", jenis_kelamin_3025)
print("Alamat saya adalah: ", alamat_3025)
print("Agama saya adalah: ", agama_3025)

Nama saya adalah:  Dika Giowanda
Jenis kelamin saya adalah:  L
Alamat saya adalah:  
    Padang Lawas
    kec. Luhak Nan Duo
    Kab. Pasaman Barat

Agama saya adalah:  islam


**Penjelasan :**
1. Inisialisasi Variabel String Baris Tunggal dengan Tanda Petik Ganda (" ")

    Variabel nama diinisialisasi dengan nilai "Dika Giowanda". Penggunaan tanda petik ganda digunakan untuk menyimpan data berupa teks (string) dalam satu baris kalimat.

2. Inisialisasi Variabel String dengan Tanda Petik Tunggal (' ')

    Variabel jenis_kelamin bernilai 'L' dan agama bernilai 'islam'. Di Python, tanda petik tunggal memiliki fungsi yang sama dengan tanda petik ganda untuk mendefinisikan tipe data String, baik untuk satu karakter maupun teks berupa kata.

3. Deklarasi Multi-line String menggunakan Triple Quotes (""" """)
    
    Variabel alamat didefinisikan menggunakan tanda petik ganda tiga kali ("""). Fitur ini berfungsi untuk membuat teks string yang mencakup beberapa baris (multiline string), di mana format baris baru (newline) dan spasi di dalamnya tetap dipertahankan saat program dijalankan.

4. Menampilkan Data Teks
    
    Fungsi print() digunakan sebanyak empat kali untuk menampilkan label teks statis yang digabungkan dengan variabel teks. Penggunaan koma (,) di dalam fungsi print() berfungsi untuk menggabungkan string sekaligus otomatis menambahkan spasi di antara label teks dan isi variabel yang dipanggil.

### 2.2.4 Program "Input_2511533025"

In [1]:
# Buat file dengan nama Input_2511533025.py
# Buat program untuk menghitung luas persegi panjang
# Nama variabel ditambah 4 digit nim terakhir contoh: panjang_123417
# Program ini menggunakan fungsi input() untuk meminta pengguna
# Memasukan nilai panjang dan lebar persegi panjang.
# Nilai yang dimasukan akan dikonversi menjadi tipe data integer
panjang_3025 = input('Masukan nilai panjang: ')
lebar_3025 = input('Masukan nilai lebar: ')
luas_3025 = int(panjang_3025) * int(lebar_3025)
print("Luas =", luas_3025)

Luas = 119


**Penjelasan :**
1. Menerima Input Dinamis dari Pengguna
    
    Fungsi input() pada variabel panjang dan lebar digunakan untuk menerima input dari pengguna melalui konsol. Secara standar (default), nilai yang ditangkap oleh fungsi input() di Python selalu bertipe data String (teks).

2. Konversi Tipe Data Eksplisit (Type Casting)
    
    Fungsi int() digunakan untuk melakukan type casting, yaitu mengubah tipe data masukan yang semula berbentuk String menjadi tipe data Integer. Hal ini wajib dilakukan karena operasi matematika tidak dapat langsung dilakukan pada tipe data String.

3. Operasi Aritmatika dan Perhitungan Luas
    
    Variabel luas menampung hasil perkalian antara int(panjang) dan int(lebar). Menggunakan operator perkalian (*), sistem menghitung luas persegi panjang berdasarkan nilai integer yang telah dikonversi (pada contoh output: 17×7=119).
    
4. Menampilkan Hasil Perhitungan (Output)

    Fungsi print("Luas =", luas) digunakan untuk mencetak label teks "Luas =" beserta nilai akhir yang tersimpan dalam variabel luas ke layar terminal.

### 2.2.5 Program "Konstanta_2511533025"

In [1]:
# Buat file dengan nama Konstanta_NIM.py
# Program ini menggunakan konstantas untuk menghitung luas lingkaran
# Nama variabel ditambah 4 digit nim terakhir contoh: jari_1234

from typing import Final
PI: Final = 3.14
print("pi: %f" % (PI))
jari_3025 = float(input('Masukan nilai jari-jari: '))
luas_3025 = PI * jari_3025 * jari_3025
print("Luas lingkaran dengan jari-jari %.2f adalah %.2f" % (jari_3025, luas_3025))

pi: 3.140000
Luas lingkaran dengan jari-jari 17.00 adalah 907.46


**Penjelasan :**
1. Import Modul dan Deklarasi Konstanta (Final)

    Perintah from typing import Final digunakan untuk mengambil penanda tipe Final. Variabel PI: Final = 3.14 mendefinisikan nilai konstanta Pi secara tetap.

2. Input dan Konversi Tipe Data Float

    Fungsi input() mengambil nilai jari-jari dari pengguna yang secara langsung dikonversi menggunakan float(...). Hal ini bertujuan agar variabel jari dapat menerima input berupa angka desimal maupun bilangan bulat untuk presisi perhitungan matematika.

3. Perhitungan Luas Lingkaran

    Variabel luas menghitung luas lingkaran berdasarkan rumus Luas=π×r2 dengan mengalikan konstanta PI dan nilai variabel jari secara berulang (PI * jari * jari).

4. Formatting Output Menggunakan String Formatting (%)
    Penggunaan operator penanda format % digunakan untuk mengatur tampilan desimal pada output:
    
    a. %f menampilkan nilai float secara default dengan 6 digit desimal (misal: 3.140000).

    b. %.2f membatasi jumlah angka di belakang koma menjadi tepat 2 digit desimal saja (misal: 17.00 dan 907.46), sehingga tampilan variabel jari dan luas menjadi lebih rapi dan terbaca.

---
## Tugas Praktikum

**Soal :**

<div align="center">
<img src="images/soal_tugas2.png" width="500"><br>
<em>Gambar 2.2 : Soal Tugas Praktikum</em>
</div>

**Kode Jawaban :**

a. Kode Program "tugas2_2511533025"

In [2]:
from typing import Final

# ==========================================
# SISTEM REGISTRASI PRAKTIKAN ALPRO 2026
# ==========================================

# Konstanta batas kelulusan
BATAS_LULUS_3025: Final[float] = 75.0

print("=== SISTEM REGISTRASI PRAKTIKAN ALPRO 2026 ===")

# ==========================================
# INPUT DATA PRAKTIKAN
# ==========================================

nama_3025 = input("Masukkan Nama Mahasiswa     : ")
jenis_kelamin_3025 = input("Masukkan Jenis Kelamin (L/P): ")
umur_3025 = int(input("Masukkan Umur               : "))
nilai_3025 = float(input("Masukkan Skor Tes Awal      : "))

# Alamat menggunakan string multiline
alamat_3025 = """Jl. Flamboyan,
Kec. Padang Lawas,
Kab. Pasaman Barat"""

# Token identifikasi menggunakan tipe Complex
id_token_sinyal_3025 = 100 + 3j

# ==========================================
# PROSES VALIDASI KELULUSAN
# ==========================================

lulus_3025 = nilai_3025 >= BATAS_LULUS_3025

# ==========================================
# MENAMPILKAN DATA DAN TIPE DATA
# ==========================================

print()
print("=== DATA PRAKTIKAN & HASIL PEMERIKSAAN ===")

print("Nama Mahasiswa :", nama_3025,
      "| Tipe:", type(nama_3025))

print("Jenis Kelamin  :", jenis_kelamin_3025,
      "| Tipe:", type(jenis_kelamin_3025))
print("Alamat Domisili:")
print(alamat_3025,
      "| Tipe:", type(alamat_3025))

print("Umur           :", umur_3025, "tahun",
      "| Tipe:", type(umur_3025))

print("Skor Tes Awal  :", nilai_3025,
      "| Tipe:", type(nilai_3025))
print("ID Token Sinyal:", id_token_sinyal_3025,
      "| Tipe:", type(id_token_sinyal_3025))

# ==========================================
# MENAMPILKAN STATUS KELULUSAN
# ==========================================

print()
print("=== STATUS KELULUSAN PRAKTIKUM ===")

print("Batas Minimum Nilai:", BATAS_LULUS_3025)
print("Apakah Dinyatakan Lulus?:",
      lulus_3025,
      "| Tipe:", type(lulus_3025))

=== SISTEM REGISTRASI PRAKTIKAN ALPRO 2026 ===

=== DATA PRAKTIKAN & HASIL PEMERIKSAAN ===
Nama Mahasiswa : Dika | Tipe: <class 'str'>
Jenis Kelamin  : L | Tipe: <class 'str'>
Alamat Domisili:
Jl. Flamboyan,
Kec. Padang Lawas,
Kab. Pasaman Barat | Tipe: <class 'str'>
Umur           : 19 tahun | Tipe: <class 'int'>
Skor Tes Awal  : 97.0 | Tipe: <class 'float'>
ID Token Sinyal: (100+3j) | Tipe: <class 'complex'>

=== STATUS KELULUSAN PRAKTIKUM ===
Batas Minimum Nilai: 75.0
Apakah Dinyatakan Lulus?: True | Tipe: <class 'bool'>


**Penjelasan:**
1. Import Modul dan Pendefinisian Konstanta (Final)

    Baris from typing import Final mengimpor tipe khusus untuk mendeklarasikan nilai tetap. Variabel BATAS_LULUS: Final[float] = 75.0 berfungsi sebagai konstanta batas minimum nilai kelulusan dengan tipe data float (bilangan desimal). Penulisan nama variabel menggunakan huruf kapital menandakan bahwa nilai tersebut bersifat tetap dan tidak boleh diubah oleh bagian program lainnya.

2. Penggunaan Fungsi input() dan Konversi Tipe Data Casting

    Program meminta masukan data dari pengguna melalui terminal menggunakan fungsi input():

    a. Variabel nama dan jenis_kelamin langsung menyimpan data dalam bentuk default, yaitu String (teks).

    b. Variabel umur menggunakan fungsi int(...) untuk mengonversi hasil input teks menjadi tipe data Integer (bilangan bulat).
    
    c. Variabel nilai menggunakan fungsi float(...) agar nilai skor tes awal yang dimasukkan dapat berupa angka desimal.

3. Inisialisasi Variabel Multi-line String dan Complex

    a. Variabel alamat dideklarasikan menggunakan tanda petik tiga (""") untuk menampung teks alamat yang terdiri dari beberapa baris (multiline string), sehingga format baris baru di dalamnya tetap terjaga.

    b. Variabel id_token_sinyal diinisialisasi dengan nilai 100 + 3j, yang merupakan bentuk penerapan tipe data Complex (bilangan kompleks dengan bagian riil dan imajiner).

4. Proses Evaluasi Logika Kelulusan

    Variabel lulus menampung hasil perbandingan antara skor tes (nilai) dengan batas minimum (BATAS_LULUS) menggunakan operator komparasi >=. Jika skor lebih besar atau sama dengan 75, maka variabel ini akan menghasilkan nilai Boolean berupa True, dan sebaliknya jika kurang akan bernilai False.

5. Pencetakan Output dan Pemeriksaan Tipe Data Dinamis (type())

    Bagian akhir program menggunakan fungsi print() untuk menampilkan seluruh data mahasiswa beserta label informasi teks. Di setiap baris cetakan data, disertakan fungsi type(...) secara berdampingan untuk membuktikan dan mencetak kelas tipe data resmi dari masing-masing variabel secara dinamis di terminal (seperti <class 'str'>, <class 'int'>, <class 'float'>, <class 'complex'>, dan <class 'bool'>).

Tahapan Commit dan Push file ke akun Github:

1. Klik tombol biru "Publish to GitHub".
2. Jika ada pop up konfirmasi atau instruksi untuk login/authorize ke akun GitHub milikmu, klik Allow dan ikuti proses login-nya di browser.
3. Pilih opsi opsi repository:
Publish to GitHub private repository (jika ingin repository hanya bisa dilihat olehmu/pengajar)
Publish to GitHub public repository (jika ingin bisa diakses siapa saja)
4. Centang file tugas dan laporan yang ingin diunggah, lalu klik OK.
5. VS Code akan otomatis melakukan Initialize, Commit pertama, dan Push ke GitHub.
6. Jika nanti kamu menambahkan atau mengedit file tugas/laporan lagi di VS Code, buka kembali tab Source Control (ikon cabang di sebelah kiri).
7. Ketik pesan perubahan pada kolom Message (contoh: "Menambahkan file laporan tugas").
8. Klik tombol Commit.
9. Klik tombol Sync Changes (atau Publish Branch / Push) yang muncul untuk mengunggah perubahan ke GitHub.

---
# BAB III
# KESIMPULAN

Berdasarkan praktikum yang telah dilakukan, dapat disimpulkan bahwa pemahaman mengenai variabel dan tipe data (seperti integer, float, complex, boolean, dan string) merupakan fondasi utama dalam membangun program yang efisien dan bebas dari error. Variabel berfungsi sebagai wadah penyimpanan nilai di memori, sedangkan tipe data menentukan jenis nilai serta operasi yang dapat diberlakukan terhadapnya.

Dalam konteks pemrograman Python, konsep ini memanfaatkan alokasi memori dinamis dan konversi tipe data eksplisit serta penggunaan konstanta, sehingga proses pengolahan input pengguna, perhitungan aritmatika, evaluasi kondisi logika, hingga pemformatan output dapat dijalankan dengan akurat dan terstruktur.

---
## DAFTAR PUSTAKA

[1] W3Schools, "Python Variables," W3Schools Online Web Tutorials, 2024. [Online]. Tersedia: https://www.w3schools.com/python/python_variables.asp. [Diakses 12 September 2026].

[2] W3Schools, "Python Data Types," W3Schools Online Web Tutorials, 2024. [Online]. Tersedia: https://www.w3schools.com/python/python_datatypes.asp. [Diakses 12 September 2026].

[3] Python Software Foundation, "Built-in Types — Python 3.12.0 Documentation," Python Documentation, 2024. [Online]. Tersedia: https://docs.python.org/3/library/stdtypes.html. [Diakses 12 September 2026].

[4] H. Enter, "Dasar-Dasar Pemrograman Python," 2020, Hal. 15–32.